main.go

```diff
@@ -0,0 +1,100 @@
+// A simple proxy that multiplexes a unix socket connection
+//
+// Copyright 2017 HyperHQ Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+package main
+
+import (
+	"flag"
+	"os"
+	"os/signal"
+	"sync"
+	"syscall"
+
+	"github.com/golang/glog"
+	"github.com/moby/moby/pkg/term"
+)
+
+func main() {
+	var (
+		debug         bool
+		logDir        string
+		root          string
+		agent         string
+		container     string
+		process       string
+		proxyExitCode bool
+		proxyStdio    bool
+		proxySignal   bool
+		proxyWinsize  bool
+	)
+
+	flag.BoolVar(&debug, "debug", false, "enable debug logging")
+	flag.StringVar(&logDir, "logdir", "", "the directory for the logging")
+	flag.StringVar(&root, "root", "", "root directory for storage of container state")
+	flag.StringVar(&agent, "agent", "", "agent gRPC socket endpoint")
+
+	flag.StringVar(&container, "container", "", "container id for the shim")
+	flag.StringVar(&process, "process", "", "process id for the shim")
+	flag.BoolVar(&proxyExitCode, "proxy-exit-code", true, "proxy exit code of the process")
+	flag.BoolVar(&proxyStdio, "proxy-stdio", true, "proxy stdio of the process")
+	flag.BoolVar(&proxySignal, "proxy-signal", true, "proxy signals to the process")
+	flag.BoolVar(&proxyWinsize, "proxy-winsize", true, "proxy tty size to the process")
```

Comment:
Not one I can think of. These are just in case.

Reply:
This list of parameters seems pretty complex. We should make this simpler. We really need to know about the container/process IDs, and the socket URL. But the rest has to be done in every case, we don't need to enable/disable such things with parameters.

Reply:
@bergwolf I propose we remove the options that are not used for now and simplify the shim CLI. It's easier to add new options when needed than to remove one and potentiall break backward compatibility.

Reply:
I think it won't hurt anything if we already have them configurable. 

Moreover, I am thinking about split `proxy-stdio` to `in` and `output`, because sometimes we don't want to input anything, for example, logs only.

Reply:
Having options just in case makes both the code base and the CLI/UI more complex. Simplicity should be an important goal for us.

Reply:
Fair enough. I'll remove unnecessary options. We can add them back when needed in future, but not in the first versions.